lib/backend/syncersv1/updateprocessors/processor.go

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) 2017 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package updateprocessors
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/projectcalico/libcalico-go/lib/backend/model"
+	"github.com/projectcalico/libcalico-go/lib/backend/watchersyncer"
+)
+
+// NewSimpleUpdateProcessor implements an update processor that only needs to take in
+// a conversion function. This is only meant to be used in cases where the Process and
+// OnSyncerStarting methods do not have any special handling.
+//
+// One of the main differences between the simple update processor and the conflict
+// resolving name cache processor is that the conflict resolving name cache processor
+// uses a cache to handle conflicts when v1 objects cannot be built up using only the
+// resource key. The simple update processor is for converting objects that can create
+// the v1 key using only the information available in the resource key.
+
+type ConvertV2ToV1Key func(v2Key model.ResourceKey) (model.Key, error)
+type ConvertV2ToV1Value func(interface{}) (interface{}, error)
```

Comment:
I believe all of our v2 reources will implement the runtime.Object interface, so the method signature could be further improved to:

type ConvertV2ToV1Value func(runtime.Object) (interface{}, error)

Reply:
I thought that the KVPair Value was set to be an interface which means that the value needs to be typecast to runtime.Object right? Should we be typecasting this to the runtime.Object interface even though we are going to typecast this to the more specific resources right after?

Reply:
I guess it doesn't really matter - I think i was thinking that it further limits misuse.

I'm not fussed if that doesn't change.